rust/common-ifc/src/graph/graph_builder.rs

```diff
@@ -0,0 +1,131 @@
+use std::{collections::HashMap, ops::Deref};
+
+const NO_NODE_FOUND: &str = "No node found.";
+
+pub struct Node<T> {
+    inner: T,
+    id: PortGraphId,
+    inputs: Vec<String>,
+    outputs: Vec<String>,
+}
+
+impl<T> Deref for Node<T> {
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        self.inner()
+    }
+}
+
+impl<T> Node<T> {
+    pub fn id(&self) -> &PortGraphId {
+        &self.id
+    }
+
+    pub fn inner(&self) -> &T {
+        &self.inner
+    }
+}
+
+/*
```

Comment:
Remove of leave a note

Reply:
This entire file shouldn't be here, oops